mlos_core/mlos_core/optimizers/bayesian_optimizers/smac_optimizer.py

```diff
@@ -0,0 +1,212 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""
+Contains the wrapper class for SMAC Bayesian optimizers.
+"""
+
+from pathlib import Path
+from typing import Mapping, Optional
+
+import ConfigSpace
+import numpy.typing as npt
+import pandas as pd
+
+from mlos_core.optimizers.bayesian_optimizers import BaseBayesianOptimizer
+from mlos_core.spaces.adapters.adapter import BaseSpaceAdapter
+
+
+class SmacOptimizer(BaseBayesianOptimizer):
+    """Wrapper class for SMAC based Bayesian optimization.
+
+    Parameters
+    ----------
+    parameter_space : ConfigSpace.ConfigurationSpace
+        The parameter space to optimize.
+
+    space_adapter : BaseSpaceAdapter
+        The space adapter class to employ for parameter space transformations.
+
+    seed : Optional[int]
+        Seed is used to make results reproducible. A value of `None` will use a random seed.
+
+    run_name : str
+        Name of this run. This is used to easily distinguish across different runs.
+
+    output_directory : str
+        The directory where SMAC output will saved.
+
+    n_random_init : Optional[int]
+        Number of points evaluated at start to bootstrap the optimizer.
+
+    n_random_probability: Optional[float]
+        Probability of choosing to evaluate a random configuration during optimization.
+        Setting this to a higher value favors exploration over exploitation.
+
+    n_workers: int
+        Number of parallel workers to use.
```

Comment:
Is this for support of simultaneous experiment trial runs, or for parallel threads during model operations?

Reply:
I assumed it was the latter, but turns out it's the former (i.e., from https://automl.github.io/SMAC3/main/advanced_usage/9_parallelism.html)

So, I guess we do not want to expose this as argument to the user and set it `1`, correct?